drogue-tls-sys/include/drogue_config.h

```diff
@@ -3540,7 +3539,7 @@
  * Uncomment to set the maximum plaintext size of both
  * incoming and outgoing I/O buffers.
  */
-#define MBEDTLS_SSL_MAX_CONTENT_LEN             1024
+//#define MBEDTLS_SSL_MAX_CONTENT_LEN             1024
```

Comment:
This will now default back to 16kb inbound and 16kb outbound buffers calloc'd from mbedtls, per connection.

I wonder if there's a happy medium between 1kb and 16kb.

Reply:
IIRC it was "number of bits" / 8 … so a length of 512 should give us 4096 keys … I don't think there are RSA keys bigger than 4k.

It would be nice to make this configurable through a Cargo feature, not sure that is possible.

Reply:
the MAX_CONTENT_LEN is for general I/O buffers, I think, and defaults to 16kb, which I've also seen allocating 2x when setting up the ssl_context.  I think the MPI stuff is maybe the keysize-related bits. We should dig in a little further.

But yes, either way, we should figure out if there's either a way to parameterize the build (since it won't affect the bindgen stuff, I don't think?) or provide some handy features of reasonable sizes (max_content_1k, max_content_4k, etc)